src/fp-paradigm/js/generate-items.js

```diff
@@ -1,6 +1,9 @@
-const generateInput = (value) => {
+// Pure functions, generates a node.
+const generateInput = (value, className) => {
     const input = document.createElement('input');
```

Comment:
Im not even sure, if we create this `const input = doc...` outside of the function and just accept it in the function will this be bad code, but isn't the fact that you're creating a variable inside the function agains OOP principles?

Reply:
Agree, it's just a simple constructor which builds an input element in our DOM, and it is pure. There we've passed input `value` and with this input, we get an output DOM Element. By the way, creating variables which are `const` are good, because they don't have any side effects that could change the output of our function. it is same to seperate `f(x) => x ** 2`, same we can decalre `x ** 2` in const variable which can't be mutated, and be passed as output from. 
**Example:**
```
 
const f = (x) => {
    const pow2 = x ** 2;
    return pow2;
}

const f = (x) => x ** 2;
```

Hope, you get my point.

Reply:
ye